btree/btree.c

```diff
@@ -3992,42 +3968,56 @@ int64_t btree_open_tree_tick(struct m0_sm_op *smop)
  */
 int64_t btree_close_tree_tick(struct m0_sm_op *smop)
 {
-	struct m0_btree_op    *bop     = M0_AMB(bop, smop, bo_op);
-	struct td             *td_curr = bop->bo_arbor->t_desc;
-	struct nd             *nd_head = ndlist_tlist_head(&td_curr->
-							   t_active_nds);
+	struct m0_btree_op *bop  = M0_AMB(bop, smop, bo_op);
+	struct td          *tree = bop->bo_arbor->t_desc;
+	struct nd          *node;
+	struct nd          *root;
+	uint64_t	    active_node_count = 0;
+
+	M0_PRE(tree->t_ref != 0);
 
 	switch (bop->bo_op.o_sm.sm_state) {
 	case P_INIT:
-		M0_ASSERT(td_curr->t_ref != 0);
-		if (td_curr->t_ref > 1) {
-			tree_put(td_curr);
+		if (tree->t_ref > 1) {
+			tree_put(tree);
 			return P_DONE;
 		}
-		td_curr->t_starttime = m0_time_now();
+		tree->t_starttime = m0_time_now();
 		/** Fallthrough to P_TIMECHECK */
 
 	case P_TIMECHECK:
 		/**
 		 * This code is meant for debugging. In future, this case needs
 		 * to be handled in a better way.
 		 */
-		if (ndlist_tlist_length(&td_curr->t_active_nds) > 1) {
-			if (m0_time_seconds(m0_time_now() -
-					    td_curr->t_starttime) > 5) {
-				td_curr->t_starttime = 0;
-				return M0_ERR(-ETIMEDOUT);
+		m0_rwlock_read_lock(&lru_lock);
+		m0_tl_for(ndlist, &btree_active_nds, node) {
+			if (node->n_tree == tree && node->n_ref > 0) {
+				root = node;
+				active_node_count++;
+				if (active_node_count > 1) {
+					if (m0_time_seconds(m0_time_now() -
+							    tree->t_starttime)
+					    > 5) {
+						m0_rwlock_read_unlock(&lru_lock);
+						tree->t_starttime = 0;
+						return M0_ERR(-ETIMEDOUT);
```

Comment:
Do not return error here. Instead crash by calling ASSERT so that the crash dump can be used for debugging if any unexpected logic is present.

Reply:
okay, I will add assert.